bokehjs/src/lib/models/annotations/info_pane.ts

```diff
@@ -0,0 +1,178 @@
+    }
+
+    const {anchor} = this.model
+    this.el.classList.remove(bk_right)
+    this.el.classList.remove(bk_left)
+    this.el.classList.remove(bk_above)
+    this.el.classList.remove(bk_below)
+
+    const arrow_size = 10  // XXX: keep in sync with less
+
+    display(this.el)
+
+   // slightly confusing: side "left" (for example) is relative to point that
+   // is being annotated but CS class ".bk-left" is relative to the tooltip itself
+    let top: number = 0
+    let left = 0
+    let right = 0
+
+    switch (anchor) {
+      case "left":
+        this.el.classList.add(bk_right)
+        left = x + (this.el.offsetWidth - this.el.clientWidth)
+        top = y - this.el.offsetHeight/2
+        break
+      case "right":
+        this.el.classList.add(bk_left)
+        right = (this.plot_view.layout.bbox.width - x)
+        top = y - this.el.offsetHeight/2
+        break
+      case "above":
+        this.el.classList.add(bk_below)
+        top = y + (this.el.offsetHeight - this.el.clientHeight) - arrow_size
+        left = Math.round(x - this.el.offsetWidth/2)
+        break
+      case "below":
+        this.el.classList.add(bk_above)
+        top = y - this.el.offsetHeight + arrow_size
+        left = Math.round(x - this.el.offsetWidth/2)
+        break
+    }
+
+    if (this.model.show_arrow)
+      this.el.classList.add(bk_tooltip_arrow)
+
+    //might be not useful in the current context
+    if (this.el.childNodes.length > 0) {
+      this.el.style.top = `${top}px`
+      this.el.style.left = left ? `${left}px` : 'auto'
+      this.el.style.right = right ? `${right}px` : 'auto'
+    } else
+      undisplay(this.el)
+  }
+}
+
+export namespace InfoPane {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Annotation.Props & {
+    anchor: p.Property<TooltipAttachment>
+    //can be ignored if inner_only property isnt useful
+    inner_only: p.Property<boolean>
+    show_arrow: p.Property<boolean>
+    x_anchor: p.Property<number>
+    y_anchor: p.Property<number>
+    //data has to be able to get list of strings
+    data: p.Property<[string][]>
+    custom: p.Property<boolean>
+  }
+}
+
+export interface InfoPane extends InfoPane.Attrs {}
+
+export class InfoPane extends Annotation {
+  properties: InfoPane.Props
+  __view_type__: InfoPaneView
+
+  constructor(attrs?: Partial<InfoPane.Attrs>) {
+    super(attrs)
+  }
+
+  static init_InfoPane(): void {
+    this.prototype.default_view = InfoPaneView
+
+    this.define<InfoPane.Props>({
+      //is kept as is if the horizontal\vertical functionality shall come in use
+      anchor:     [ p.TooltipAttachment, 'horizontal' ],
+      inner_only: [ p.Boolean,           true         ],
+      show_arrow: [ p.Boolean,           true         ],
+      x_anchor:   [ p.Number,             0           ],
+      y_anchor:   [ p.Number,             0           ],
+    })
+
+    this.override({
+      level: 'overlay',
+    })
+
+    this.internal({
+      data:   [ p.Any, [] ],
```

Comment:
Why define this as internal? Also can you explain the reasoning behind making this an array? Not saying it's wrong I just don't immediately see the benefits.

Reply:
Could be made a String as well. But I think as we tend to use it for Hovertool in future and array shall do a better job. As we can get more than one panes on a single glyph.

Reply:
This should definitely be made public, i.e. moved out of `this.internal` and in to `this.define` above. "Internal" properties are only for the rare cases where we need a property *only* on the JS side. But that's no longer the case now that we are adding a Python side `InfoPane` model with `data` property. 

As for the type, I think it is String or list of String. I will leave a comment in the main PR body about that discussion